src/snowflake/cli/api/utils/definition_rendering.py

```diff
@@ -232,6 +239,26 @@ def _validate_env_section(env_section: dict):
             )
 
 
+def _get_referenced_vars_in_definition(
+    template_env: TemplatedEnvironment, definition: Definition
+):
+    referenced_vars = set()
+
+    def find_any_template_vars(element):
+        referenced_vars.update(template_env.get_referenced_vars(element))
+
+    traverse(definition, visit_action=find_any_template_vars)
+
+    return referenced_vars
+
+
+def _template_version_warning():
```

Comment:
I still think that with non-PDF files templating coming soon, this should take in the file name and use that instead of a generic "project definition file" description. This way when we template more files we can have uniform error messages. Thoughts?

Reply:
This code is special for project definition file (due to the chicken and egg issue with it). The rest of templating will use simpler templating and they can produce their own messages. Once we move to pdf v2, we can then easily have access to the file name since we won't have 2 merged files anymore. For now, project definition file might be a good compromise (since it matches what's in the documentation - https://docs.snowflake.com/en/developer-guide/snowflake-cli-v2/project-definitions/about)

Reply:
I was after not changing the pattern of the message for the very next feature we'll implement. I generally prefer specific file names in warnings/errors as opposed to vague descriptions that require bouncing to the documentation and back. I won't insist though, do as you see fit.